beacon-chain/rpc/beaconv1/validator.go

```diff
@@ -175,70 +185,81 @@ func valContainersByRequestIds(state iface.BeaconState, validatorIds [][]byte) (
 				valIndex = types.ValidatorIndex(index)
 			}
 			v1Validator := migration.V1Alpha1ValidatorToV1(allValidators[valIndex])
+			subStatus, err := validatorSubStatus(v1Validator, epoch)
+			if err != nil {
+				return nil, fmt.Errorf("could not get validator sub status: %v", err)
+			}
 			valContainers[i] = &ethpb.ValidatorContainer{
 				Index:     valIndex,
 				Balance:   allBalances[valIndex],
-				Status:    validatorSubStatus(v1Validator, epoch),
+				Status:    subStatus,
 				Validator: v1Validator,
 			}
 		}
 	}
 	return valContainers, nil
 }
 
-func validatorStatus(validator *ethpb.Validator, epoch types.Epoch) ethpb.ValidatorStatus {
-	switch validatorSubStatus(validator, epoch) {
+func validatorStatus(validator *ethpb.Validator, epoch types.Epoch) (ethpb.ValidatorStatus, error) {
+	valStatus, err := validatorSubStatus(validator, epoch)
+	if err != nil {
+		return 0, errors.Wrap(err, "could not get sub status")
+	}
+	switch valStatus {
 	case ethpb.ValidatorStatus_PENDING_INITIALIZED, ethpb.ValidatorStatus_PENDING_QUEUED:
-		return ethpb.ValidatorStatus_PENDING
+		return ethpb.ValidatorStatus_PENDING, nil
 	case ethpb.ValidatorStatus_ACTIVE_ONGOING, ethpb.ValidatorStatus_ACTIVE_SLASHED, ethpb.ValidatorStatus_ACTIVE_EXITING:
-		return ethpb.ValidatorStatus_ACTIVE
+		return ethpb.ValidatorStatus_ACTIVE, nil
 	case ethpb.ValidatorStatus_EXITED_UNSLASHED, ethpb.ValidatorStatus_EXITED_SLASHED:
-		return ethpb.ValidatorStatus_EXITED
+		return ethpb.ValidatorStatus_EXITED, nil
 	case ethpb.ValidatorStatus_WITHDRAWAL_POSSIBLE, ethpb.ValidatorStatus_WITHDRAWAL_DONE:
-		return ethpb.ValidatorStatus_WITHDRAWAL
+		return ethpb.ValidatorStatus_WITHDRAWAL, nil
 	}
-	return 0
+	return 0, errors.New("invalid validator state")
 }
 
-func validatorSubStatus(validator *ethpb.Validator, epoch types.Epoch) ethpb.ValidatorStatus {
+func validatorSubStatus(validator *ethpb.Validator, epoch types.Epoch) (ethpb.ValidatorStatus, error) {
+	if validator == nil {
+		return 0, errors.New("validator is nil")
+	}
```

Comment:
I presume you added this to have an error test scenario because it's probably impossible to set up an invalid state. If yes, then I would not do it. We normally don't validate input parameters like this (whether that's OK is a different story), and adding code just to fulfill a test is usually an anti-pattern. I think it's OK to expect no errors in every test case.

Reply:
Added it since it would cause a panic if it wasnt there and validator was nil, but maybe it isn't needed. Will remove.